docs/posts/ibis-overturemaps/index.qmd

```diff
@@ -0,0 +1,213 @@
+```
+
+Now that we have the data lets explore it in Ibis interactive mode and make some beautiful maps.
+
+## Data exploration
+
+```{python}
+import ibis
+from ibis import _
+
+ibis.options.interactive = True
+con = ibis.get_backend() # default duckdb backend
+```
+
+```{python}
+usa_infra = con.read_parquet("usa-infra.geoparquet")
+usa_infra
+```
+
+We take a look at the subtypes of infrastructure:
+
+```{python}
+usa_infra.subtype.value_counts().preview(max_rows=15)
+```
+
+If we want to look at power lines we need to look into `subtype=="power"`
+and see what kind of `class` we have in there.
+
+```{python}
+(usa_infra.filter(_.subtype=="power")["class"]
+          .value_counts()
+          .order_by(ibis.desc("class_count"))
+          )
+```
+
+Looks like we have `plants`, `power_lines` and `minor_lines`, so we can make get some nice maps.
+
+```{python}
+plants = usa_infra.filter(_.subtype=="power", usa_infra["class"]=="plant")
+power_lines = usa_infra.filter(_.subtype=="power", usa_infra["class"]=="power_line")
+minor_lines = usa_infra.filter(_.subtype=="power", usa_infra["class"]=="minor_line")
+```
+
+
+## Plotting
+
+**Note: maybe here explain why lonboard ?**
+
+::: {.callout-note}
+You can try this in your machine, for the purpose the blog file size, we will show
+screenshots of the visualization
+:::
+
+```python
+import lonboard
+from lonboard.basemap import CartoBasemap # to choose color of basemap
+```
+
+Let's visualize the `power plants`
+
+```python
+lonboard.viz(plants,
+            scatterplot_kwargs={"get_fill_color": "red"},
+            polygon_kwargs={"get_fill_color": "red"},
+            map_kwargs={"basemap_style": CartoBasemap.Positron,
+            "view_state": {"longitude": -100, "latitude": 36, "zoom": 3}
+            })
+```
+
+![Power plants in the USA](usa-power-plants.png)
+
+If you are visualizing this in your machine, you can zoom in and see some of the
+geometry where the plants are located. But as an example, we can plot in a small
+area of California:
+
+```python
+plants_CA = plants.filter(_.bbox.xmin.between(-118.6, -117.9),
+           _.bbox.ymin.between(34.5, 35.3))[_.names.primary, _.geometry]
+```
+
+```python
+lonboard.viz(plants_CA,
+            scatterplot_kwargs={"get_fill_color": "red"},
+            polygon_kwargs={"get_fill_color": "red"},
+            map_kwargs={"basemap_style": CartoBasemap.Positron,
+            })
+```
+
+![Power plants near Lancaster CA](ca-power-plants.png)
+
+We can also visualize together the `power_lines` and the `minor_lines` by doing:
+
+
+```python
+lonboard.viz([minor_lines, power_lines])
+```
+
+![Minor and Power lines of USA](usa-power-and-minor-lines.png)
+
+and that's how you can visualize ~7 million points from the comfort of
+your laptop.
+
+Note: I got the ~7M by adding the number of points in power_lines and minor_lines. I'm not sure if plotting the lines that connect these points add points to this.
```

Comment:
It is more effort for the GPU to render lines than to render points, but a line isn't rendered as a collection of points.

It might be clearer to say `7 million coordinates` rather than points.